core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/iterator/GroupIterator.java

```diff
@@ -129,7 +133,13 @@ public GroupIterator(EvaluationStrategy strategy, Group group, BindingSet parent
 
 	@Override
 	public void handleClose() throws QueryEvaluationException {
-		cf.close();
+		try {
+			cf.close();
+		} finally {
+			if (argumentsIter != null) {
+				argumentsIter.close();
```

Comment:
Cant argumwntsIter still be null? It could be set to null between this line and the null check above.

Reply:
Yes, indeed. Missed that in your initial comment.